test/e2e/pod_restart_test.go

```diff
@@ -0,0 +1,165 @@
+	})
+
+	It("podman pod restart all pods", func() {
+		session := podmanTest.Podman([]string{"pod", "create", "--name", "foobar99"})
+		session.WaitWithDefaultTimeout()
+
+		session = podmanTest.RunTopContainerInPod("test1", "foobar99")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session2 := podmanTest.Podman([]string{"pod", "create", "--name", "foobar100"})
+		session2.WaitWithDefaultTimeout()
+
+		session = podmanTest.RunTopContainerInPod("test2", "foobar100")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		startTime := podmanTest.Podman([]string{"inspect", "--format='{{.State.StartedAt}}'", "test1", "test2"})
+		startTime.WaitWithDefaultTimeout()
+
+		session = podmanTest.Podman([]string{"pod", "restart", "-a"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		restartTime := podmanTest.Podman([]string{"inspect", "--format='{{.State.StartedAt}}'", "test1", "test2"})
+		restartTime.WaitWithDefaultTimeout()
+		Expect(restartTime.OutputToStringArray()[0]).To(Not(Equal(startTime.OutputToStringArray()[0])))
+		Expect(restartTime.OutputToStringArray()[1]).To(Not(Equal(startTime.OutputToStringArray()[1])))
+	})
+
+	It("podman pod restart latest pod", func() {
+		session := podmanTest.Podman([]string{"pod", "create", "--name", "foobar99"})
+		session.WaitWithDefaultTimeout()
+
+		session = podmanTest.RunTopContainerInPod("test1", "foobar99")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session2 := podmanTest.Podman([]string{"pod", "create", "--name", "foobar100"})
+		session2.WaitWithDefaultTimeout()
+
+		session = podmanTest.RunTopContainerInPod("test2", "foobar100")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		startTime := podmanTest.Podman([]string{"inspect", "--format='{{.State.StartedAt}}'", "test1", "test2"})
+		startTime.WaitWithDefaultTimeout()
+
+		session = podmanTest.Podman([]string{"pod", "restart", "-l"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		restartTime := podmanTest.Podman([]string{"inspect", "--format='{{.State.StartedAt}}'", "test1", "test2"})
+		restartTime.WaitWithDefaultTimeout()
+		Expect(restartTime.OutputToStringArray()[0]).To(Equal(startTime.OutputToStringArray()[0]))
+		Expect(restartTime.OutputToStringArray()[1]).To(Not(Equal(startTime.OutputToStringArray()[1])))
+	})
+
+	It("podman pod restart multiple pods with bogus", func() {
+		session := podmanTest.Podman([]string{"pod", "create", "--name", "foobar99"})
+		session.WaitWithDefaultTimeout()
+		cid1 := session.OutputToString()
+
+		session = podmanTest.RunTopContainerInPod("", "foobar99")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"pod", "restart", cid1, "doesnotexist"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(125))
+	})
+})
```

Comment:
Maybe I missed it due to a lack of tea, but a test that restarted a pod with three or more containers in it would be good to have.